rewiringamerica/api.rewiringamerica.org
# Convert Program TS files to JSON. Update script to generate JSON from spreadsheet data.

Convert Program TS files to JSON. Update script to generate JSON from spreadsheet data.
This is done to make the data files more consistent which will aid us in how we integrate incentive admin data modifications back into the API. See: The [Mini Brief](https://docs.google.com/document/d/1nM-4Utwbe68tcY60Rw94a9HZQp1l5PWobkQdSEqUQfI/edit?usp=sharing) for more details. 

## Links

- [_Asana_](https://app.asana.com/0/1207304265252553/1207827064980175/f)
- [Incentive Admin Mini-brief](https://docs.google.com/document/d/1nM-4Utwbe68tcY60Rw94a9HZQp1l5PWobkQdSEqUQfI/edit?usp=sharing)

## Changes

-  Convert state_programs.ts files to JSON
- update scripts/lib/authority-and-program-updater.ts to generate JSON files when executing 
- update README

## Test Plan

- run `yarn ts-node scripts/generate-misc-state-data.ts [TWO CHAR STATE CODE]` and verify corresponding `state/programs.json` file is created. 
- run `yarn dev` 
- hit the api with a request for a SLU incentive ie: `http://localhost:3000/api/v1/calculator?zip=89015&include_beta_states=true&owner_status=homeowner&household_income=2000&tax_filing=single&household_size=1` 

## Next steps

- Update Import and Export workflow in Incentive Admin.
